src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala

```diff
@@ -11,11 +11,55 @@ import org.apache.spark.sql.sources.InsertableRelation
 import org.apache.spark.sql.types.{DataTypes, StructType}
 import org.apache.spark.sql.{Row, SQLContext}
 import com.cognite.spark.datasource.SparkSchemaHelper._
+import io.circe.Decoder.Result
+import io.circe.{Decoder, Encoder, HCursor, Json}
 import org.apache.spark.datasource.MetricsSource
 
 import scala.concurrent.ExecutionContext
 
-case class PostAssetsDataItems[A](items: Seq[A])
+sealed case class Type(id: Long, name: String, fields: Seq[FieldTypes.Field])
+
+object FieldTypes {
+  case class Field(id: Long, name: String, valueType: String, value: String)
+  case class DoubleIsTooLargeForJSON(field: String)
+      extends Throwable(s"Double is too large for JSON in data type field $field.")
+
+  implicit val fieldDecoder: Decoder[Field] = new Decoder[Field] {
+    override def apply(c: HCursor): Result[Field] =
+      for {
+        id <- c.downField("id").as[Long]
+        name <- c.downField("name").as[String]
+        valueType <- c.downField("valueType").as[String]
+        value <- valueType match {
+          case "String" => c.downField("value").as[String]
+          case "Long" => c.downField("value").as[Long].map(_.toString)
+          case "Double" => c.downField("value").as[Double].map(_.toString)
+          case "Boolean" => c.downField("value").as[Boolean].map(_.toString)
+        }
```

Comment:
I wonder what the error message looks like if it doesn't match one of the cases. Maybe we should add a default case to get an informative error?

It's possible that more value types might be added in the future. The perfect scenario would be if there was some generic "toString" method in circe.